calico_cni_k8s_test.go

```diff
@@ -309,6 +309,93 @@ var _ = Describe("CalicoCni", func() {
 					}))
 				})
 			})
+
+			Context("using ipAddrs annotation to assign IP address to a pod, through Calico IPAM", func() {
```

Comment:
I would favour a little more testing here. Could this be made into a table driven test that tests for example a single IPv4 address, a single IPv6 address, mixed IPv4 and IPv6 and error cases like too many addresses?